### ISSUE TITLE
Already tired of cpu high up bug

### ISSUE THREAD

**Issue by elpedriyo** [greatman/craftconomy3#95]
The plugin highs up my cpu 50% because of this plugin, could you please solve it?
Here are the images:
My normal cpu- 
![image](https://cloud.githubusercontent.com/assets/3207791/14547671/68b72534-02b0-11e6-98e4-26db6c348841.png)
The high cpu- 
![image](https://cloud.githubusercontent.com/assets/3207791/14547673/728cb4ac-02b0-11e6-97a6-26312293f8d7.png)

Please

**Comment by greatman** [comment]
Can you try to run this: https://raw.githubusercontent.com/major/MySQLTuner-perl/master/mysqltuner.pl ?

It will tell you if MySQL needs some tuning (cache, table open, etc) which can improves your CPU usage. Since I use a lot of keys /foreign keys it may improve performance quite a bit.

**Comment by elpedriyo** [comment]
how do I run that?

**Comment by greatman** [comment]
It's a perl script, just wget it then do perl mysqltuner.pl

**Comment by elpedriyo** [comment]
and do you recommend me to execute it when I have the cpu more than 50% because of mysql?

**Comment by elpedriyo** [comment]
I also where monitoring task in mysql and I saw it is the select process from your plugin the problem, but I am going to run this now, to see what can I do

**Comment by elpedriyo** [comment]
I got this when cpu was fine:
![image](https://cloud.githubusercontent.com/assets/3207791/14551128/838e0dea-02cf-11e6-9fa4-a524ef66a67d.png)

let me see now with it high

**Comment by elpedriyo** [comment]
Ok, the other image when the cpu was up was the same

**Comment by greatman** [comment]
The script only asks that the mysql server has been running for more than
24h.

**Comment by elpedriyo** [comment]
Ok, so then with the results you see, do you recommed something for optimizing this?
Because when your plugin launches the select query my cpu goes up to a 100% sometimes and this is horrible for all servers on the machine :/

**Comment by elpedriyo** [comment]
Also it tells that the plugin is not closing mysql connections properly :/

**Comment by greatman** [comment]
Yeah well the script use the mysql stats it sees so having a large pool of data (aka at least 24h running) will provide somewhat accurate informations. Then we will see what is causing the issues. Sometimes a default MySQL config isn't enough for some workloads.

**Comment by elpedriyo** [comment]
Ok, so then tomorrow I will paste you again what I see.
Another thing, at the startup of my servers I have always an error from craftconomy, could you please solve that?

**Comment by elpedriyo** [comment]
Also when I introduce table_cache value mysql do not start again, I do not know if it is compatible with mysql 5.7

**Comment by elpedriyo** [comment]
@greatman It is the same, the performance continue going bad when the plugin performs this:
![image](https://cloud.githubusercontent.com/assets/3207791/14551715/e507f796-02d6-11e6-8b98-a3ab1b4cd31a.png)

Could you simply optimize the command?

**Comment by elpedriyo** [comment]
The problem persists, it is a problem with the select function of the plugin every 3 or 4 minutes each time. Do not know why the plugin is making select function each 3 or 4 mintues

**Comment by elpedriyo** [comment]
Ok, any solution, could you fix it?

**Comment by elpedriyo** [comment]
I continue having the same problems

**Comment by netherstar** [comment]
is this issue is fixed in latest version?
